airflow/providers/cncf/kubernetes/operators/pod.py

```diff
@@ -191,6 +191,8 @@ class KubernetesPodOperator(BaseOperator):
     :param is_delete_operator_pod: What to do when the pod reaches its final
         state, or the execution is interrupted. If True (default), delete the
         pod; if False, leave the pod.
+    :param delete_when_fails: If False, and is_delete_operator_pod is set to True
```

Comment:
This feels a bit complicated.
I think we should consider replace is_delete_operator with:
action_on_finish that supports several modes. For example: delete_pod, delete_pod_success

That way you dont need to work hard to understand the combinationa of flags, the mode is self explanatory.

WDYT?

Reply:
My first thought was doing something similar to #29394, but I hesitated to deprecate `is_delete_operator_pod`, but yes, these flags may confuse users. I'll implement the argument you propose.

Reply:
cc @jedcunningham WDYT?

Reply:
Yep. Agree with @eladkal simple flag is better. I made similar "keep old flag" approaches in the past and it led to really bad complexity/